neurodamus/ngv.py

```diff
@@ -11,65 +10,114 @@
+        """
         super().__init__(gid, meinfos, None)
 
-        # Compose the path to the morphology file
-        morph_file = (
+        # Create the cell
+        self._cellref = Nd.Cell(gid)  # cell instantiated with Cell.hoc
+        # load and apply morphology
+        morph = MorphIOWrapper(
             Path(circuit_conf.MorphologyPath)
             / f"{meinfos.morph_name}.{circuit_conf.MorphologyType}"
         )
-
-        # Create the cell and load morphology
-        self._cellref = Nd.Cell(gid)
-        morph = MorphIOWrapper(morph_file)
         self._cellref.AddHocMorph(morph.morph_as_hoc())
-
-        self._nseg_warning = 0
-
         # Recalculate number of segments and sections
         self._cellref.geom_nseg_fixed()
         self._cellref.geom_nsec()
 
         logging.debug("Instantiating NGV cell gid=%d", gid)
 
-        # Insert mechanisms and glutamate receptors in each section
-        self.sections_glut = []
-        for sec in self._cellref.all:
-            if sec.nseg > 1:
-                self._nseg_warning = 1
-                sec.nseg = 1
-            sec.insert("cadifus")
-            glut = Nd.GlutReceive(sec(0.5), sec=sec)
-            Nd.setpointer(glut._ref_glut, "glu2", sec(0.5).cadifus)
-            self.sections_glut.append(glut)
-
-        # Configure endoplasmic reticulum and section parameters
-        self._cellref.execute_commands(self._er_as_hoc(morph))
-        self._cellref.execute_commands(self._secparams_as_hoc(morph))
-
-        # Soma-specific glutamate receptor (must be last)
-        # used only for accounting for metabolsim. It should not be
-        # connected to other point processes or mechanisms
+        # assigned later
+        self._gluts = {}
+        is_resized_secs = False
+        for sec in self.all:
+            is_resized = self._init_basic_section(sec)
+            is_resized_secs = is_resized_secs or is_resized
+
+        if is_resized_secs:
+            logging.warning(
+                "The astrocyte %s had some of their sections in `all` "
+                "reduced to 1 compartment. Multi-compartment "
+                "sections are not supported at the moment.",
+                self.gid,
+            )
+
+        # add GlutReceiveSoma (only for metabolism)
         soma = self._cellref.soma[0]
-        self.soma_glut = Nd.GlutReceiveSoma(soma(0.5), sec=soma)
+        self.glut_soma = Nd.GlutReceiveSoma(soma(0.5), sec=soma)
 
-        self._cellref.gid = gid
+        self.gid = gid
         self.section_names = morph.section_names
 
+    def _get_sec(self, sec_id):
+        """Retrieve section object by section ID."""
+        section_name = self.section_names[sec_id]
+        sec_list = getattr(self._cellref, section_name.name)
+        return sec_list[section_name.id]
+
+    @staticmethod
+    def _init_basic_section(sec):
+        """Init a normal section
+
+        Initialize sec with mechanisms and point processes
+        (store PP to avoid GC).
+        """
```

Comment:
Can you add what is returned in the docstring? Becuase it is not clear to me from the function name.

Reply:
good point. Done